library/std/src/Std/DurrHoyerLibrary.qs

```diff
@@ -0,0 +1,207 @@
+import Std.Math.*;
+import Std.Arrays.*;
+import Std.Convert.*;
+
+function CountElements(list : Int[], threshold : Int, comparisonType : String) : Int {
```

Comment:
This function can be simplified by passing in a comparison operator to avoid type=="min"/"max" in code.

Please refer to this [Playground sample](https://microsoft.github.io/qsharp/?code=H4sIAAAAAAAACoVRTUvDQBC951c8gocEt9SKCjZuBMVDD70InkoPMW7Mwma2ZDdQifnvstmYbvHgHAbmzZt581F1VFqpCc%252B6I%252FuiRCPImkRJY7HGhuxuz2DrVphaqw8PMZS6ORStNJqwRrIhu8jxpLVKU89AHwFA09niXQmUrjc4rrJoxCvdQngpSMIo5iucySron0y8NCA4M8JObS85VtmcG5wbo8FrtcJ2LXluFg1RVP1uvC0kJW7gN5L%252FTayE9WNy7FYM1wy3DHcM9%252FuAcDoTx02Ifx0EOOJGUuwnXS7PE8VxSqhxrfm4fKJMxXjEcZHj%252BHBS%252BvZIPiOT7lYYU3yK5CJ%252BFQYc%252Fd%252F%252FBn8NP5oObHyQU%252Ba98wNsXdDZev0cDHGaRe6uP3MibiNKAgAA&profile=unrestricted) for more details.